client/src/error.rs

```diff
@@ -0,0 +1,97 @@
+use std::str;
+
+use hyper;
+use serde_json;
+
+#[derive(Debug, Fail)]
+pub enum ClientError {
+    /// What was being attempted when the decode error occured.
+    // https://github.com/rust-lang-nursery/failure/issues/183 prevents
+    // easily deferring rendering until needed.
```

Comment:
Oh that's annoying.  We can't "manually" implement Display for just this one enum variant either :(